core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java

```diff
@@ -468,20 +470,27 @@ public void setSelectedExpression(Expression expression) {
     }
   }
 
+  private double clampValue(double min, double max, double val) {
+    return Math.min(max, Math.max(min, val));
+  }
+
   private void updateCameraView(EntityImp targetImplementation) {
     AxisAlignedBox alignedBox = targetImplementation.getDynamicAxisAlignedMinimumBoundingBox(org.lgna.story.implementation.AsSeenBy.SCENE);
     double targetHeight = alignedBox.getHeight();
     double targetWidth = alignedBox.getWidth();
     double targetDepth = alignedBox.getDepth();
-    double defaultDistance = 1.0f;
     AffineMatrix4x4 targetTransform = targetImplementation.getAbsoluteTransformation();
     ClippedZPlane picturePlane = new ClippedZPlane();
 
     // Update layout camera marker
     AffineMatrix4x4 layoutTransform = AffineMatrix4x4.createIdentity();
+    double CAMERA_CLAMP_MAX = 100.0;
+    double CAMERA_CLAMP_MIN = 2.0;
+    double PICTURE_CLAMP_MAX = 100.0;
+    double PICTURE_CLAMP_MIN = 1.0;
     layoutTransform.translation.x = targetTransform.translation.x;
-    layoutTransform.translation.y = targetTransform.translation.y + DEFAULT_LAYOUT_CAMERA_Y_OFFSET;
-    layoutTransform.translation.z = targetTransform.translation.z - DEFAULT_LAYOUT_CAMERA_Z_OFFSET;
+    layoutTransform.translation.y = targetTransform.translation.y + (targetHeight != 0 ? clampValue(CAMERA_CLAMP_MIN, CAMERA_CLAMP_MAX, targetHeight * DEFAULT_LAYOUT_CAMERA_Y_OFFSET) : DEFAULT_LAYOUT_CAMERA_Y_OFFSET);
+    layoutTransform.translation.z = targetTransform.translation.z - (targetHeight != 0 ? clampValue(CAMERA_CLAMP_MIN, CAMERA_CLAMP_MAX, targetHeight * DEFAULT_LAYOUT_CAMERA_Z_OFFSET) : DEFAULT_LAYOUT_CAMERA_Z_OFFSET);
```

Comment:
are you intentionally using height and z together here?  if so, can you add a comment about why this makes sense?

Reply:
yes, I've tried the depth value but the height value is more accurate in calculating the z-offset. I think that is because I'm offsetting the camera with the distance of a square.